core/infra/config/iroha_config_with_json.cpp

```diff
@@ -1,50 +1,30 @@
-/*
-Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+#include "iroha_config_with_json.hpp"
 
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
+using IrohaConfigManager = config::IrohaConfigManager;
 
-     http://www.apache.org/licenses/LICENSE-2.0
+IrohaConfigManager::IrohaConfigManager(){}
 
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
+IrohaConfigManager& IrohaConfigManager::getInstance() {
+    static IrohaConfigManager instance;
+    return instance;
+}
 
-#include "iroha_config_with_json.hpp"
+std::string IrohaConfigManager::getConfigName() {
+  return "config/config.json";
+}
+
+std::string IrohaConfigManager::getDatabasePath(const std::string& defaultValue) {
+  return getParam("database_path", defaultValue);
+}
+
+size_t IrohaConfigManager::getConcurrency(size_t defaultValue) {
+  return getParam("concurrency", defaultValue);
+}
+
+size_t IrohaConfigManager::getMaxFaultyPeers(size_t defaultValue) {
+  return getParam("maxFaultyPeers", defaultValue);
+}
 
-namespace config {
-    IrohaConfigManager::IrohaConfigManager() { }
-
-    std::string IrohaConfigManager::openJSONText(const std::string& PathToJSONFile) {
```

Comment:
Logic of opening/reading from file should not be duplicated/overwritten in derived classes.